api/src/main/java/io/grpc/Status.java

```diff
@@ -662,8 +658,6 @@ private static String parseAsciiStringSlow(byte[] value) {
    */
   @Override
   public boolean equals(Object obj) {
-    assert !FAIL_ON_EQUALS_FOR_TEST
-        : "Status.equals called; disable this by setting " + TEST_EQUALS_FAILURE_PROPERTY;
```

Comment:
Seems like a big change. I'm pretty sure I hit error when was comparing statuses in unit tests, and it forced me to compare them manually.

Reply:
I did a grep and couldn't find any usages. So that meant to me it wasn't used. If you can find a use, I'll keep it.

Reply:
That's kinda the point - this error is what prevented them.

Reply:
Discussed IRL. This is not enabled anyway.